PathOfBuildingCommunity/PathOfBuilding
# Add support for displaying config for chance based mods only when necessary

When in combat:
- "Do you have Onslaught?"
- "Do you have Arcane Surge?"
- "Do you have Unholy Might?"
- "Are you Fortified?"
- "Do you have Tailwind?"
- "Do you have Adrenaline?"
- "Are you on Consecrated Ground?"
- "Are you Chilled?"

For effective dps:
- "Is the enemy Intimidated?"
- "Is the enemy Unnerved?"
- "Is the enemy Crushed?"
- "Is the enemy Blinded?"
- "Is the enemy Covered in Ash?"
- "Is the enemy Covered in Frost?"

This also adds parsing for Recoup leech mastery as I thought I will need it for Tanu Ahi but decided to make it just enabled the 2 checkboxes instead.

Sources of chance to fortify are parsed and simply allow displaying checkbox. Same with blind/unnerve/intimidate.

Sources of Consecrated Ground are parsed and are adding Conditional entry for OnConsecratedGround that simply shows the checkbox.

Same applies for chance-based and unreliable sources of Adrenaline and Onslaught, they use new "Conditional" condition.

Changed the condition for displaying is enemy
poisoned/bleeding/chilled/shocked/ignited to check if you can inflict that ailment and only then display. This also allows me to remove the workaround for showing is enemy ignited added before.

### Steps taken to verify a working solution:
- Check conditional sources of Onslaught/Adrenaline/Unholy Might/Tailwind/Fortify/Crush/Intimidate/Unnerve/Blind/Ash/Frost/Consecratd Ground
- Verify that checkbox appears

### Link to a build that showcases this PR:
https://pobb.in/mFj-CFwnN-45

### Before screenshot (Empty pob):
![image](https://user-images.githubusercontent.com/5115805/235375152-1eb207c6-de13-4e0f-81f2-e5d43c3894c5.png)

### After screenshot (Empty pob):
![image](https://user-images.githubusercontent.com/5115805/235375119-7e8f2a31-1670-48dd-85d7-d162b15f5ef0.png)
